examples/sql/sql_order_by_limit_offset.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <hazelcast/client/hazelcast_client.h>
+
+int
+main()
+{
+    using namespace hazelcast::client::sql;
+
+    auto hz = hazelcast::new_client().get();
+
+    // populate the map with some data
+    auto map = hz.get_map("myMap").get();
+    map->put("key1", 1.0);
+    map->put("key2", 2.0);
+    map->put("key3", 3.0);
+    map->put("key4", 4.0);
+    map->put("key5", 5.0);
+
+    auto sql = hz.get_sql();
+    // Create mapping for the integers. This needs to be done only once per map.
+    auto result = sql
+                    .execute(R"(
+                      CREATE OR REPLACE MAPPING myMap (
+                          __key VARCHAR,
+                          this DOUBLE
+                      )
+                      TYPE IMAP
+                      OPTIONS (
+                          'keyFormat' = 'varchar',
+                          'valueFormat' = 'double'
+                      )
+                    )")
+                    .get();
+
+    // Fetch values in between (40, 50)
+    result =
+      sql
+        .execute("SELECT * FROM myMap ORDER BY this ASC LIMIT ? OFFSET ?", 3, 1)
```

Comment:
Please put more documentation at the start of the file and also at the important lines explaining what those lines do, what are expected results, et. The more doc, the more useful it is for the user.